plugins/hydra_joblib_launcher/hydra_plugins/hydra_joblib_launcher/joblib_launcher.py

```diff
@@ -76,20 +76,25 @@ def launch(self, job_overrides: Sequence[Sequence[str]]) -> Sequence[JobReturn]:
         configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
         sweep_dir = Path(str(self.config.hydra.sweep.dir))
         sweep_dir.mkdir(parents=True, exist_ok=True)
+
+        # Joblib's backend is hard-coded to loky since the threading
+        # backend is incompatible with Hydra
+        joblib_keywords = OmegaConf.to_container(self.joblib, resolve=True)
+        joblib_keywords["backend"] = "loky"
```

Comment:
If I may, just a comment: you should warn the user that keyword arg `backend` is bypassed if it is set explicitely by the user.

There is other process based backends than `loky` for `joblib` such as `dask`, `hadoop` and others. Bypassing an explicit without warning will create weird behavior and might confuse some users.

Reply:
This is not overriding but preventing the user from set the backend  at all from the config.
```
>>> from omegaconf import OmegaConf
>>> c1 = OmegaConf.create({"foo": 10})
>>> OmegaConf.set_struct(c1, True)
>>> OmegaConf.merge(c1, {"foo": 20})
{'foo': 20}

>>> OmegaConf.merge(c1, {"foo": 20, "backend": "loki"})
...
AttributeError: Accessing unknown key in a struct : backend
```
The concern is that there are issues with other backends as well that we do not have the time to test right now.
I agree with you that it's best to just block the threading backend and allow users to play with the others.

If you have some cycles, It would be great if you can help testing some of the other backends.
If they need some additional config we will probably need to change the plugin a bit to support it.

Reply:
I agree that this is a good idea to restrict the backend to `loky`. My only concern is the fact that this is silent if the user give a different `backend` argument. I would say just keep this behavior but raise a warning. I believe no one will see it but just in case, it will avoid some troubles in debugging as it will be clear why the backend is changed back to `loky` :)

Reply:
Ah, so, the current version of the PR takes out `backend` from the plugin config altogether. In effect, users cannot provide a `backend` argument to the plugin at all. `loky` will always be used. 

Alternatively, we might offer `backend` in the config and raise an exception if set to `threading`. However, I'm not sure this offers any benefits, also see my reply here: https://github.com/facebookresearch/hydra/pull/423#discussion_r381162808

Reply:
Nevermind then if users cannot pass the argument directly!
Thanks for your explanation.